src/common/utils/media-capabilities.js

```diff
@@ -0,0 +1,143 @@
+      isDRMSupported: SupportedOptions.NOT_SUPPORTED,
+      supportedDRMs: []
+    };
+    _logger.debug('Returning media capabilities defaults ', {mediaCapabilities});
+    return mediaCapabilities;
+  }
+}
+
+/**
+ * checks whether and which DRMs are supported by the browser
+ * @function _checkDRMSupported
+ * @returns {Promise<DRMSupportedObject>} - The DRM supported object.
+ * @private
+ */
+async function _checkDRMSupported(): Promise<DRMSupportedObject> {
+  const drmSupportedRes: DRMSupportedObject = {
+    isDRMSupported: SupportedOptions.MAYBE_SUPPORTED,
+    supportedDRMs: []
+  };
+
+  if (!navigator.requestMediaKeySystemAccess) {
+    return drmSupportedRes;
+  }
+
+  const keySysConfig = [
+    {
+      initDataTypes: ['cenc'],
+      videoCapabilities: [
+        {
+          contentType: CONTENT_TYPE_AVC_CODEC
+        }
+      ]
+    }
+  ];
+
+  const keySystemsMap = new Map(DRM_SCHEME_LIST);
+
+  for (let [drm, keySys] of keySystemsMap) {
+    await navigator
+      .requestMediaKeySystemAccess(keySys, keySysConfig)
+      .then(() => {
+        drmSupportedRes.supportedDRMs.push(drm);
+      })
+      .catch(function (ex) {
+        _logger.debug(keySys + ' not supported (' + ex.name + ': ' + ex.message + ').');
+      });
+  }
+
+  drmSupportedRes.isDRMSupported = drmSupportedRes.supportedDRMs.length > 0 ? SupportedOptions.SUPPORTED : SupportedOptions.NOT_SUPPORTED;
+  return drmSupportedRes;
+}
+
+/**
+ * checks whether the browser supports HEVC codec or not
+ * @function _checkHEVCSupported
+ * @param {HEVCConfigObject} hevcConfig - The HEVC configuration.
+ * @returns {Promise<HEVCSupportedObject>} - The HEVC supported object.
+ * @private
+ */
+async function _checkHEVCSupported(hevcConfig?: HEVCConfigObject): Promise<HEVCSupportedObject> {
+  const hevcSupportedRes: HEVCSupportedObject = {
+    isHEVCSupported: SupportedOptions.MAYBE_SUPPORTED,
+    isPowerEfficient: SupportedOptions.MAYBE_SUPPORTED
+  };
+
+  if (!navigator.mediaCapabilities || !navigator.mediaCapabilities.decodingInfo) {
+    return hevcSupportedRes;
+  }
+
+  const configHvc = {
+    type: 'media-source',
+    video: {
+      contentType: CONTENT_TYPE_HVC_CODEC,
+      width: hevcConfig?.width || WIDTH_DEFAULT,
+      height: hevcConfig?.height || HEIGHT_DEFAULT,
+      bitrate: hevcConfig?.bitrate || BITRATE_DEFAULT,
+      framerate: hevcConfig?.framerate || FRAMERATE_DEFAULT
+    }
+  };
+
+  await navigator.mediaCapabilities
+    .decodingInfo(configHvc)
+    .then(result => {
+      hevcSupportedRes.isHEVCSupported = result.supported ? SupportedOptions.SUPPORTED : SupportedOptions.NOT_SUPPORTED;
+      hevcSupportedRes.isPowerEfficient = result.powerEfficient ? SupportedOptions.SUPPORTED : SupportedOptions.NOT_SUPPORTED;
+    })
+    .catch(() => {
+      hevcSupportedRes.isHEVCSupported = SupportedOptions.NOT_SUPPORTED;
+      hevcSupportedRes.isPowerEfficient = SupportedOptions.NOT_SUPPORTED;
+    });
+  return hevcSupportedRes;
```

Comment:
`hevcSupportedRes` is sent before is it set at line 133/134

Reply:
@JonathanTGold it is not.. I just double checked. also, if that was the case, it would always return MAYBE, but it is returning the right value from my testing- SUPPORTED.